plugins/languages/treesitter/rainbow-delimiters.nix

```diff
@@ -0,0 +1,212 @@
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
+with lib; let
+  helpers = import ../../helpers.nix {inherit lib;};
+in {
+  # TODO those warnings have been added XX/XX/2023
+  # -> Remove them in ~ 1 month (oct. 2023)
+  imports =
+    mapAttrsToList
+    (
+      old: new:
+        mkRenamedOptionModule
+        ["plugins" "treesitter-rainbow" old]
+        ["plugins" "rainbow-delimiters" new]
+    )
+    {
+      enable = "enable";
+      package = "package";
+      strategy = "strategy";
+      query = "query";
+      disable = "blacklist";
+      hlgroups = "highlight";
+    };
+
+  options.plugins.rainbow-delimiters =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "rainbow-delimiters.nvim";
+
+      package =
+        helpers.mkPackageOption
+        "rainbow-delimiters.nvim"
+        pkgs.vimPlugins.rainbow-delimiters-nvim;
+
+      strategy =
+        helpers.defaultNullOpts.mkNullable
+        (with types; attrsOf str)
+        ''
+          {
+            default = "require 'rainbow-delimiters'.strategy.global";
+          }
+        ''
+        ''
+          Attrs mapping Tree-sitter language names to strategies.
+          See `|rb-delimiters-strategy|` for more information about strategies.
+
+          Example:
+          ```nix
+            {
+              # Use global strategy by default
+              default = "require 'rainbow-delimiters'.strategy['global']";
+
+              # Use local for HTML
+              html = "require 'rainbow_delimiters'.strategy['local']";
+
+              # Pick the strategy for LaTeX dynamically based on the buffer size
+              latex = \'\'
+                function()
+                  -- Disabled for very large files, global strategy for large files,
+                  -- local strategy otherwise
+                  if vim.fn.line('$') > 10000 then
+                      return nil
+                  elseif vim.fn.line('$') > 1000 then
+                      return rainbow.strategy['global']
+                  end
+                  return rainbow.strategy['local']
+                end
+              \'\';
+            }
+          ```
+        '';
+
+      query =
+        helpers.defaultNullOpts.mkNullable
+        (with types; attrsOf str)
```

Comment:
We _could_ be more helpfull by only listing valid queries (https://github.com/HiPhish/rainbow-delimiters.nvim/blob/9cbd3dc409af1f5531778ccd1ea6bce668241f39/doc/rainbow-delimiters.txt#L302-L319) but I feel this would make the code too complicated

Reply:
Yes, I am not sure that this would add a lot of value...